deadlines/src/main/java/com/palantir/deadlines/Deadlines.java

```diff
@@ -112,30 +112,23 @@ public static void disableFurtherDeadlinePropagation() {
      * this value and one already set via a previous call to {@link #parseFromRequest}, if it exists
      * @param request the request object to write the encoding to
      * @param adapter a {@link RequestEncodingAdapter} that handles writing the header value to the request object
+     * @param enforcement a client requested {@link Enforcement} state, which will be resolved against any existing state
      */
     public static <T> void encodeToRequest(
-            Duration proposedDeadline, T request, RequestEncodingAdapter<? super T> adapter) {
+            Duration proposedDeadline, T request, RequestEncodingAdapter<? super T> adapter, Enforcement enforcement) {
```

Comment:
i think we should maintain backwards compatibility by adding an override with the previous method signature, but it's not clear what `Enforcement` value would be used in that case to delegate to this method.

perhaps this needs to be an `Optional<Enforcement`>? I'm not sure what would make the most sense. I think the logic we're going for in that case is "caller has no opinion about the enforcement strategy", and we should likely just use the previous logic. wdyt?

Reply:
side note: it looks like this repository is not using revapi (my fault!) but it most definitely should be! this is an API break.